src/neptune/internal/operation_processors/async_operation_processor.py

```diff
@@ -178,10 +185,17 @@ def _wait_for_queue_empty(self, initial_queue_size: int, seconds: Optional[float
                     max_reconnect_wait_time,
                 )
             else:
-                logger.warning(
-                    "Waiting for the remaining %s operations to synchronize with Neptune." " Do not kill this process.",
-                    initial_queue_size,
-                )
+                if msg_queue is not None:
+                    msg_queue.put(initial_queue_size)
+                else:
+                    logger.warning(
```

Comment:
I think that we should move this logging part somewhere for deduplication. Maybe this could be a separate methods or even a class.